go/ipset/api.go

```diff
@@ -0,0 +1,284 @@
+}
+
+func handlePassthrough(res http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	IP := vars["ip"]
+	Port := vars["port"]
+	Local := vars["local"]
+
+	jobs <- job{"Add", "pfsession_passthrough", IP + "," + Port}
+	if Local == "0" {
+		updateClusterPassthrough(IP, Port)
+	}
+	var result = map[string][]*Info{
+		"result": {
+			&Info{IP: IP, Status: "ACK"},
+		},
+	}
+
+	res.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	res.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(res).Encode(result); err != nil {
+		panic(err)
+	}
+
+}
+
+func handleIsolationPassthrough(res http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	IP := vars["ip"]
+	Port := vars["port"]
+	Local := vars["local"]
+
+	jobs <- job{"Add", "pfsession_isol_passthrough", IP + "," + Port}
+	if Local == "0" {
+		updateClusterPassthroughIsol(IP, Port)
+	}
+	var result = map[string][]*Info{
+		"result": {
+			&Info{IP: IP, Status: "ACK"},
+		},
+	}
+
+	res.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	res.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(res).Encode(result); err != nil {
+		panic(err)
+	}
+
+}
+
+func handleLayer2(res http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	IP := vars["ip"]
+	Mac := vars["mac"]
+	Network := vars["network"]
+	Type := vars["type"]
+	Catid := vars["catid"]
+	Local := vars["local"]
+
+	// Update locally
+	IPSET.IPSEThandleLayer2(IP, Mac, Network, Type, Catid)
+
+	// Do we have to update the other members of the cluster
+	if Local == "0" {
+		updateClusterL2(IP, Mac, Network, Type, Catid)
+	}
+	var result = map[string][]*Info{
+		"result": {
+			&Info{Mac: Mac, Status: "ACK"},
+		},
+	}
+
+	res.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	res.WriteHeader(http.StatusOK)
+	if err := json.NewEncoder(res).Encode(result); err != nil {
+		panic(err)
+	}
+}
+
+func (IPSET *pfIPSET) IPSEThandleLayer2(IP string, Mac string, Network string, Type string, Catid string) {
+
+	for _, v := range IPSET.ListALL {
+		// Delete all entries with the new ip address
+		r := ipset.Test(v.Name, IP)
+		if r == nil {
+			jobs <- job{"Del", v.Name, IP}
+			fmt.Println("Removed " + IP + " from " + v.Name)
+		}
+		// Delete all entries with old ip addresses
+		Ips := IPSET.mac2ip(Mac, v)
+		for _, i := range Ips {
+			jobs <- job{"Del", v.Name, i}
+			fmt.Println("Removed " + i + " from " + v.Name)
+		}
+	}
+	// Add to the new ipset session
+	jobs <- job{"Add", "pfsession_" + Type + "_" + Network, IP + "," + Mac}
+	fmt.Println("Added " + IP + " " + Mac + " to pfsession_" + Type + "_" + Network)
+	if Type == "Reg" {
+		// Add to the ip ipset session
+		jobs <- job{"Add", "PF-iL2_ID" + Catid + "_" + Network, IP}
+		fmt.Println("Added " + IP + " to PF-iL2_ID" + Catid + "_" + Network)
+	}
+}
+
+func handleMarkIpL2(res http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	IP := vars["ip"]
+	Network := vars["network"]
+	Catid := vars["catid"]
```

Comment:
![image](https://user-images.githubusercontent.com/3857942/35592273-760c44a6-05da-11e8-924c-60dd90805dfa.png)

would be better named category_id